### PR TITLE
TapBreadcrumbDataSource created

### PR DESCRIPTION
**Pull request by nelsitoPuglisi**
## Goal

Create a TapBreadcrumbDataSource to log taps as span events.

## Testing

Unit test

### CODE REVIEW

**Comment by nelsitoPuglisi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/663?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#663** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/663?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#662** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/662?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nelsitoPuglisi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/SchemaType.kt [hunk 1]
```diff
@@ -71,6 +93,7 @@ internal sealed class SchemaType(
 internal object SchemaDefaultName {
     internal const val CUSTOM_BREADCRUMB = "emb-custom-breadcrumb"
     internal const val VIEW_BREADCRUMB = "screen-view"
+    internal const val TAP_BREADCRUMB = "emb-ui-tap"
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/SchemaType.kt:96]: No need to add the `emb` - it should be added for you when the span is created.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/SchemaType.kt [hunk 2]
```diff
@@ -39,6 +40,27 @@ internal sealed class SchemaType(
         override val attrs = mapOf("view.name" to viewName)
     }
 
+    /**
+     * Represents a tap breadcrumb event.
+     * @param viewName The name of the view that the tap event occurred in.
+     * @param type The type of tap event. "tap"/"long_press". "tap" is the default.
+     * @param coords The coordinates of the tap event.
+     */
+    internal class TapBreadcrumb(
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/SchemaType.kt:49]: We're trying to not use "breadcrumb" for anything other than a custom breadcrumb, which will just be called breadcrumb. Maybe just call this `Tap`?

Comment (bidetofevil) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/SchemaType.kt:49]: I'll change the rest to conform to the new name so don't worry about those haha

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/SchemaType.kt [hunk 3]
```diff
@@ -39,6 +40,27 @@ internal sealed class SchemaType(
...
+        type: String = "tap",
+        coords: String
+    ) : SchemaType(
+        EmbType.Ux.Tap,
+        TAP_BREADCRUMB
+    ) {
+        override val attrs = mapOf(
+            "view.name" to viewName,
+            "tap.type" to type,
+            "tap.coords" to coords
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/SchemaType.kt:60]: Not for the migration now, but something to think about for the future when we want to make this a semantic convention for all of OTel - do we want to have a "unit" field so tap coordinates can be in % of the viewport, device pixels, device independent pixels, etc?

**Review by bidetofevil — APPROVED** [review]
Looks good. I assume the integration tests be added in a separate PR?

**Comment by nelsitoPuglisi** [comment]
> Looks good. I assume the integration tests be added in a separate PR?

Yes, there are part of the next PR, when it's hooked up in the breadcrumb service.

**Pull request merged**